bin/src/hickory-dns.rs

```diff
@@ -333,147 +366,182 @@ fn main() {
         default();
     }
 
-    info!("Hickory DNS {} starting", hickory_client::version());
-    // start up the server for listening
+    info!("Hickory DNS {} starting...", hickory_client::version());
+
+    // Load configuration files
 
     let config = args.config.clone();
     let config_path = Path::new(&config);
-    info!("loading configuration from: {:?}", config_path);
-    let config = Config::read_config(config_path)
-        .unwrap_or_else(|e| panic!("could not read config {}: {:?}", config_path.display(), e));
+
+    info!("Loading configuration from: {config_path:?}");
+
+    let config = Config::read_config(config_path).expect("Valid configuration files");
     let directory_config = config.get_directory().to_path_buf();
     let zonedir = args.zonedir.clone();
     let zone_dir: PathBuf = zonedir
         .as_ref()
         .map(PathBuf::from)
         .unwrap_or_else(|| directory_config.clone());
 
-    // TODO: allow for num threads configured...
-    let mut runtime = runtime::Builder::new_multi_thread()
-        .enable_all()
-        .worker_threads(4)
-        .thread_name("hickory-server-runtime")
-        .build()
-        .expect("failed to initialize Tokio Runtime");
+    let mut runtime = runtime::Builder::new_multi_thread();
+    runtime.enable_all().thread_name("hickory-server-runtime");
+    if let Some(workers) = args.workers {
+        runtime.worker_threads(workers);
+    }
+    let mut runtime = runtime.build().expect("Running Tokio runtime");
+
     let mut catalog: Catalog = Catalog::new();
     // configure our server based on the config_path
     for zone in config.get_zones() {
         let zone_name = zone
             .get_zone()
-            .unwrap_or_else(|_| panic!("bad zone name in {:?}", config_path));
+            .expect("Valid zone definition in configuration file");
```

Comment:
The `config_path` will not print out the bad zone file path either. `config_path` will already be printed at the start of the app and printing it out again is redundant:

`info!("Loading configuration from: {config_path:?}");`

Reply:
By introducing the validation feature that prints out all the information in INFO stage, it may not be necessary to have redundant messaging style.

If also something went wrong, users should run the command at `--debug` stage for troubleshooting.

If you are still insisted in changes, please let me know.

Reply:
Yes, I still think we should have somewhat redundant messaging on error paths. We're optimizing (to a large extent) for making it easy to diagnose the failure, and having some redundancy in error messages imposes very limited cost.